### ISSUE TITLE
[CSV Type provider] are the type provider supposed to load and return lines in order?

### ISSUE THREAD

**Issue by mastoj** [fsharp/FSharp.Data#1046]
I loaded a file of ~100MB and it reported an error on line X. I did a lookup on line X and a couple of lines before and after but couldn't see any error. When I parsed line X manually with the type provider it passed. Also, when iterating over the `Rows` property it doesn't look like the ordering is the same as when I just iterating over the lines in the file.

Can I expect the `Rows` to be in the same order as in the file?

**Comment by ovatsus** [comment]
Yes, they should be in the same order

**Comment by mastoj** [comment]
@ovatsus, that is really strange. Maybe the provider is doing some kind of "look ahead"? That might explain what I experienced today. 

I can't share the data since that is internal to us, but what I did over a 100mb file was just a simple `Load` of the data. After that I started iterating over the seq just printing each row, and after a while it crashed. I did get a row number for when it crashed but the data on the row or the previous rows didn't match what I saw when just iterating over the file without parsing it.
